components/settings/RequestHandler.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import Dialog from 'material-ui/Dialog/Dialog';
+import FlatButton from 'material-ui/FlatButton/FlatButton';
+import RaisedButton from 'material-ui/RaisedButton/RaisedButton';
+import Utils from '../../Utils';
+
+export default class RequestHandler extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            open: this.props.open,
+            actionRequired: false,
+            errorMessage: undefined,
+            errorDetails: undefined,
+            spinning: false
```

Comment:
it used to avoid spinning and dialogbox overlapping. I am not sure after the changes, it is still required or you managed to solve the overlapping. will check =)

Reply:
something got broken @adrianq  here. I tried to load a model from a JSON file but then I get a grey screen, I can not test the purpose of the spinning state. This one will take me some time to sort out. Will update in a day or two